chsdi/templates/htmlpopup/windatlas50.mako

```diff
@@ -0,0 +1,1003 @@
+<%inherit file="base.mako"/>
+<%!
+from chsdi.lib.raster.georaster import get_raster
+
+r = get_raster('COMB')
+%>
+
+
+<%def name="table_body(c, lang)">
+<% c['stable_id'] = True %>
+<%
+from gatilegrid.grid import Grid
+from chsdi.models.grid import get_grid_spec
+
+altitude = int(c['layerBodId'].split('ch.bfe.windenergie-geschwindigkeit_h')[1])
+
+# center coordinates and dhm altitude from feature id
+gridSpec = get_grid_spec(c['layerBodId'])
+col, row = [int(x) for x in c['featureId'].split("_")]
+grid = Grid(gridSpec.get('extent'), gridSpec.get('resolutionX'), gridSpec.get('resolutionY'))
+extent = grid.cellExtent(col,row)
+center = [(extent[0] + extent[2])/2,(extent[1] + extent[3])/2]
+dhm_altitude = h.filter_alt(r.getVal(center[0], center[1]))
+center = ', '.join([str(round(center[0], 2)), str(round(center[1], 2))])
+
+props = c['properties']
+baseUrl = request.registry.settings['api_url']
+%>
+<!-- html output -->
+    <tr>
+      <th class="cell-left">${_('tt_bfe_hoehe_ueber_grund')}</th>
+      <td>${altitude or '-'}</td>
+    </tr>
+    <tr>
+      <th class="cell-left">${_('tt_bfe_koordinaten')}</th>
+      <td>${center}</td>
+    </tr>
+    <tr>
+      <th class="cell-left">${_('tt_bfe_hoehe_gelaende')}</th>
+      <td>${dhm_altitude or '-'}</td>
+    </tr>
+    <tr>
+      <th class="cell-left">${_('tt_bfe_geschw_wind_durchschnitt')}</th>
+      <td>${round(props['v_mean'], 2)}</td>
+    </tr>
+    <tr>
+      <th colspan=2>
+        <iframe src="${baseUrl}/rest/services/all/MapServer/${c['layerBodId']}/${c['featureId']}/extendedHtmlPopup?lang=${lang}&iframe=true" width="100%" height="230" frameborder="0" style="border: 0" scrolling="no"></iframe>
+      </th>
+    </tr>
+</%def>
+
+
+<%def name="extended_info(c, lang)">
+<%
+coordinates = c['geometry']['coordinates'][0]
+bottomLeft = coordinates[0]
+topRight = coordinates[2]
+center = [(bottomLeft[0] + topRight[0]) / 2, (bottomLeft[1] + topRight[1]) / 2]
+dhm_altitude = h.filter_alt(r.getVal(center[0], center[1]))
+center = ', '.join([str(round(center[0], 2)), str(round(center[1], 2))])
+altitude = int(c['layerBodId'].split('ch.bfe.windenergie-geschwindigkeit_h')[1])
+
+props = c['properties']
+
+freq_total = props['freq_0']    + props['freq_30']
+freq_total += props['freq_60']  + props['freq_90']
+freq_total += props['freq_120'] + props['freq_150']
+freq_total += props['freq_180'] + props['freq_210']
+freq_total += props['freq_240'] + props['freq_270']
+freq_total += props['freq_300'] + props['freq_330']
+
+# gracefully check if url request has get parameter iframe
```

Comment:
Yes after this PR. It's already a lot of changes without adding a new service. But I like the idea.